Multisig/Logic/UpdateController.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  AppUpdateManager.swift
+//  Multisig
+//
+//  Created by Moaaz on 4/16/21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+class UpdateController {
+    func appUpdateStatus() -> UpdateAppViewController? {
+        let remoteConfig = FirebaseRemoteConfig.shared
+        let appVersion = App.configuration.app.marketingVersion
+
+        let latestAppVersion: String? = remoteConfig.value(key: .newestVersion)
+
+        if appVersion == latestAppVersion { return nil }
+        let deprecatedVersions: String? = remoteConfig.value(key: .deprecated)
+        var style = UpdateAppViewController.Style.optional
+        if let deprecatedVersionsRange = deprecatedVersions, check(value: appVersion, in: deprecatedVersionsRange) {
+            style = .required
+        } else {
+            style = .recommended
+        }
+
+        return UpdateAppViewController(style: style)
+    }
+
+    private func check(value: String, in range: String) -> Bool {
+        let ranges = range.split(separator: ",")
+        for range in ranges {
+            let minVersion = range.components(separatedBy: "..")[0]
+            let maxVersion = range.components(separatedBy: "..")[1]
```

Comment:
Did you change the requirements for the range syntax? This implementation is not correct per requirements:

  - `x1.y1.z1-x2.y2.z2,x3.y3.z3-x4.y4.z4,x5.y5.z5`
    - This encodes 2 ranges and 1 explicit version:
    - From `x1.y1.z1 `to `x2.y2.z2`
    - From `x3.y3.z3` to `x4.y4.z4`
    - Just `x5.y5.z5`

The separator must be minus sign "-"

Also, what if there's no minus sign? The range list can be just one version number and not a lower/upper bound.